subsys/net/ip/tcp.c

```diff
@@ -417,7 +417,9 @@ int net_tcp_prepare_segment(struct net_tcp *tcp, u8_t flags,
 
 	if (flags & NET_TCP_FIN) {
 		tcp->flags |= NET_TCP_FINAL_SENT;
-		seq++;
+printk("Making sure FIN goes with ACK; adjusting seq by 2; now fire up Wireshark and check that all's ok\n");
+		flags |= NET_TCP_ACK;
+		seq += 2;
```

Comment:
This seq+=2 looks indeed weird. At least we could add a comment about why we do it like this. I agree with @pfalcon  that it does not change much if we fix this according to wireshark output.